### PR TITLE
docs: settings description of ignoreRegExpList

### PR DESCRIPTION
**Pull request by PabloLION**
fix https://github.com/streetsidesoftware/vscode-spell-checker/issues/1486

### CODE REVIEW

File: packages/_server/src/config/cspellConfig.ts [hunk 1]
```diff
@@ -599,6 +599,25 @@ interface CSpellSettingsPackageProperties extends CSpellSettings {
...
+     * - `/["href"]/`: exclude the word "href", for its frequent use in HTML.
+     * - `/\b[A-Z]{1,5}\b/g`: exclude full-caps acronyms with 1-5 length.
+     * 
+     * **JSON Example**
+     * 
+     * `.vscode/settings.json` file, `cSpell.ignoreRegExpList`
+     * accepts an array of strings with vscode escaped chars, where
+     * `\` escape to `\\`, like in `.code-snippets` files.
+     * 
+     * - `/\b[A-Z]{1,5}\b/g` in GUI becomes `/\\b[A-Z]{1,5}\\b/g` in JSON.
```
Comment (Jason3S) [inline packages/_server/src/config/cspellConfig.ts:620]: I think it reads better if you give the JSON example followed by the GUI example.

File: packages/_server/src/config/cspellConfig.ts [hunk 2]
```diff
@@ -599,6 +599,25 @@ interface CSpellSettingsPackageProperties extends CSpellSettings {
...
     /**
      * @scope resource
+     * @description
+     * @markdownDescription
+     * List of RegExp patterns or Pattern names to exclude from spell checking.
+     * The default RegExp flag is `//gi`
+     * 
+     * **GUI Examples**
+     * 
+     * Everyline is a RegExp pattern or a Pattern names - comment
```
Comment (Jason3S) [inline packages/_server/src/config/cspellConfig.ts:609]: Here is the list of predefined patterns:
```
-  `Base64`, `Base64MultiLine`, `Base64SingleLine`, `CStyleComment`, `CStyleHexValue`, 
   `CSSHexValue`, `CommitHash`, `CommitHashLink`, `Email`, `EscapeCharacters`, 
   `HexValues`, `href`, `PhpHereDoc`, `PublicKey`, `RsaCert`, `SshRsa`, `SHA`, 
   `HashStrings`, `string`, `UnicodeRef`, `Urls`, `UUID`
```

File: packages/_server/src/config/cspellConfig.ts [hunk 3]
```diff
@@ -599,6 +599,25 @@ interface CSpellSettingsPackageProperties extends CSpellSettings {
...
+     * 
+     * - `/["href"]/`: exclude the word "href", for its frequent use in HTML.
```
[the remark this answers is not in the document]
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts:611]: I agree on this. After you pointing it out, I find it's not actually a RegExp. I'll use some RegExp. Thanks.

File: packages/_server/src/config/cspellConfig.ts [hunk 4]
```diff
@@ -599,6 +599,25 @@ interface CSpellSettingsPackageProperties extends CSpellSettings {
 
     /**
      * @scope resource
+     * @description
+     * @markdownDescription
+     * List of RegExp patterns or Pattern names to exclude from spell checking.
+     * The default RegExp flag is `//gi`
+     * 
+     * **GUI Examples**
```
[the remark this answers is not in the document]
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts:607]: Thanks

File: packages/_server/src/config/cspellConfig.ts (hunk 1 above)
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts:620]: Changed to a table

File: packages/_server/src/config/cspellConfig.ts (hunk 2 above)
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts:609]: I used `venv`

Comment (Jason3S) [inline packages/_server/src/config/cspellConfig.ts:609]: `Everyline` should be two words.

**Comment by PabloLION** [comment]
Two Questions:

1. I cannot new like, the `\n` compiles to `\\n`, and although `<br/>` or `&nbsp;` compiles to `\n` but it's not rendered in the GUI.
2. The designing. We are mixing functions from `cSpell.userWords`, `Pattern name`, `RegExp Pattern`. It's better to make it clearer
    - plain words without `//` delimiter will be always considered as `Pattern name`.
    - if user wants an `RegExp Pattern`, the `//` is required.

File: packages/_server/src/config/cspellConfig.ts [hunk 5]
```diff
@@ -691,9 +710,9 @@ interface CSpellSettingsPackageProperties extends CSpellSettings {
  */
 type GlobDefX = GlobDef;
 
-export interface CustomDictionaryWithScope extends CustomDictionary {}
+export interface CustomDictionaryWithScope extends CustomDictionary { }
```
Comment (Jason3S) [inline packages/_server/src/config/cspellConfig.ts:713]: If you run `yarn lint` is should fix any issues.

**Review by Jason3S — CHANGES_REQUESTED** [review]

File: packages/_server/src/config/cspellConfig.ts (hunk 5 above)
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts:713]: Will do it this time. A suggestion would be using git hook, or, add it to the "postinstall".

File: packages/_server/src/config/cspellConfig.ts (hunk 2 above)
Comment (PabloLION) [reply packages/_server/src/config/cspellConfig.ts]: Fixed.

**Comment by Jason3S** [comment]
> Two Questions:
> 
> 1. I cannot make line break, the `\n` compiles to `\\n`, and although `<br/>` or `&nbsp;` compiles to `\n` but it's not rendered in the GUI.

Two empty lines become a visual line break.

image

Results in:

image


> 2. The designing. We are mixing functions from `cSpell.userWords`, `Pattern name`, `RegExp Pattern`. It's better to make it clearer
>    
>    * plain words without `//` delimiter will be always considered as `Pattern name`.
>    * if user wants an `RegExp Pattern`, the `//` is required.
> 

This is a bit of a legacy issue. It would be a major breaking change for a lot of repositories to enforce that. But, like you say, we can suggest how to write them in a way that hints at the right thing.

> Picture for(1) image

Interesting, I have never tried a table before.

**Comment by PabloLION** [comment]
> Two empty lines become a visual line break.

I tested this first, but its not working in the GUI. I saw it before in the `description`and `markdownDescription` are different.  and the markdown version doesn't allow line break. I don't know how do it. 

> major breaking change

I agree. Although we are adding nothing new. The forced delimiter `//gi` will compulse users to change old "RegExp pattern" . In case you agree the design is improvable, we can do a mild change: support both version but change it on the next major version update. Or even better, if we can touch user's setting file, do it programmatically.

For now, I think we can add the suggestion to the description.

**Comment by Jason3S** [comment]
#1761 includes this PR.

**Comment by PabloLION** [comment]
Thanks!

**Pull request closed without merging**